bot/constants.py

```diff
@@ -618,6 +618,17 @@ class _Icons(EnvConfig):
 Icons = _Icons()
 
 
+class _GithubModsRepository(EnvConfig):
+    section = "github_mods_"
+
+    owner: str = "python-discord"
+    name: str = "mods"
+    token: str
```

Comment:
This PR throws an error when the github token is missing (when it tries it get it from `.env` file as defined in constants). Maybe we can try making this optional. We also need to update the .env guide on the site https://www.pythondiscord.com/pages/guides/pydis-guides/contributing/bot/#appendix-full-env-file-options
```suggestion
    token: str = ""
```

Reply:
Ha ! Yes.

It didn't before because we weren't using pydantic at that time.

Good catch, i will take care of it!

Reply:
we already have `API_KEYS_GITHUB` as optional, is the new `token` constant any different or does it use a different auth approach ?

Reply:
Github introduced tokens per repo.

So this one will be granular in regards to the repo where tasks will be created

Reply:
ah okay, cool